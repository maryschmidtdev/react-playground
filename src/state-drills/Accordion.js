import React from "react";

const sectionsProps = [
  { title: "Section 1", content: "Lorem ipsum dolor" },
  { tilte: "Section 2", content: "Cupiditate tenetur aliquam" },
  { title: "Section 3", content: "Animi amet cumque sint" },
];

class Accordion extends React.Component {
  static defaultProps = { sections: [] };
  state = { currentSectionsIndex: 0 };

  handleButtonClick = (index) => {
    this.setState({ currentSectionsIndex: index });
  };

  renderButtons() {
    return this.props.sections.map((section, index) => (
      <button key={index} onClick={() => this.handleButtonClick(index)}>
        {sectionsProps.title}
      </button>
    ));
  }
  renderContent() {
    const currentSections = this.props.sections[
      this.state.currentSectionsIndex
    ];
    return <div className="content">{currentSections.content}</div>;
  }
  render() {
    return (
      <div>
        <ul>
          <li>
            {this.renderButtons()}
            {!!this.props.sections.length && this.renderContent()}
          </li>
        </ul>
      </div>
    );
  }
}

export default Accordion;
