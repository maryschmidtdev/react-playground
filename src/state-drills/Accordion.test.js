import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import Accordion from "./Accordion";

const sectionsProp = [
  { title: "Section 1", content: "Lorem ipsum dolor" },
  { tilte: "Section 2", content: "Cupiditate tenetur aliquam" },
  { title: "Section 3", content: "Animi amet cumque sint" },
];

describe("Accordion Component", () => {
  it("renders empty given no Accordion without errors", () => {
    const wrapper = shallow(<Accordion />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders no sections by default", () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("opens any clicked section", () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />);
    wrapper.find("button").at(1).simulate("click");
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("only opens one section at a time", () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />);
    wrapper.find("button").at(1).simulate("click");
    wrapper.find("button").at(2).simulate("click");
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
