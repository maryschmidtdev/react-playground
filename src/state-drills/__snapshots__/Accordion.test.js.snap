// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component only opens one section at a time 1`] = `
<div>
  <ul>
    <li>
      <button
        key="0"
        onClick={[Function]}
      />
      <button
        key="1"
        onClick={[Function]}
      />
      <button
        key="2"
        onClick={[Function]}
      />
      <div
        className="content"
      >
        Animi amet cumque sint
      </div>
    </li>
  </ul>
</div>
`;

exports[`Accordion Component opens any clicked section 1`] = `
<div>
  <ul>
    <li>
      <button
        key="0"
        onClick={[Function]}
      />
      <button
        key="1"
        onClick={[Function]}
      />
      <button
        key="2"
        onClick={[Function]}
      />
      <div
        className="content"
      >
        Cupiditate tenetur aliquam
      </div>
    </li>
  </ul>
</div>
`;

exports[`Accordion Component renders empty given no Accordion without errors 1`] = `
<div>
  <ul>
    <li />
  </ul>
</div>
`;

exports[`Accordion Component renders no sections by default 1`] = `
<div>
  <ul>
    <li>
      <button
        key="0"
        onClick={[Function]}
      />
      <button
        key="1"
        onClick={[Function]}
      />
      <button
        key="2"
        onClick={[Function]}
      />
      <div
        className="content"
      >
        Lorem ipsum dolor
      </div>
    </li>
  </ul>
</div>
`;
